feature/navigation/drawer/src/main/kotlin/app/k9mail/feature/navigation/drawer/ui/DrawerContent.kt

```diff
@@ -27,11 +27,12 @@ internal fun DrawerContent(
             .fillMaxSize()
             .testTag("DrawerContent"),
     ) {
+        val selectedAccount = state.accounts.firstOrNull { it.uuid == state.selectedAccountUuid }
```

Comment:
The problem is that the selection could happen at a point in time where the list of accounts is not propagated yet.

I think the whole selection state handling needs to be redesigned to work properly and not the responsibility of the `MessageList`.

Reply:
I could use remember to avoid calculating that value all the time.

Reply:
I suggested this because it would have kept the code diff to a minimum, i.e. the UI code could have continued to reference `State.selectedAccount`. And if we wanted to optimize the search for the account, we could do so in one place.

```kotlin
data class State(…) {
    val selectedAccount: DisplayAccount? by lazy {
        accounts.firstOrNull { it.uuid == selectedAccountUuid }
    }

    // TODO: Do same for `selectedFolder`
}
```

Since you've already changed all of the code, it doesn't matter to much. Changing it back is work, too. And there's not a lot of benefit to it. Just something to keep in mind for similar refactorings in the future.